Sources/PrimerSDK/Classes/User Interface/Root/PrimerVaultManagerViewController.swift

```diff
@@ -29,77 +29,47 @@ internal class PrimerVaultManagerViewController: PrimerFormViewController {
-                
-                switch paymentMethod.type {
-                case .paymentCard:
-                    paymentMethodButton.setTitleColor(theme.colorTheme.text1, for: .normal)
-                    paymentMethodButton.tintColor = theme.colorTheme.text1 // We want the card icon colot to be the same color as the text
-                    paymentMethodButton.layer.borderWidth = 1.0
-                    paymentMethodButton.layer.borderColor = theme.colorTheme.text1.cgColor
-                    paymentMethodButton.addTarget(self, action: #selector(cardButtonTapped), for: .touchUpInside)
-                    verticalStackView.addArrangedSubview(paymentMethodButton)
-                    
-                case .payPal:
-                    if #available(iOS 11.0, *) {
-                        paymentMethodButton.backgroundColor = UIColor(red: 0.745, green: 0.894, blue: 0.996, alpha: 1)
-                        paymentMethodButton.setTitleColor(.white, for: .normal)
-                        paymentMethodButton.setImage(paymentMethod.toIconName()?.image, for: .normal)
-                        paymentMethodButton.tintColor = .white
-                        paymentMethodButton.addTarget(self, action: #selector(payPalButtonTapped), for: .touchUpInside)
-                        verticalStackView.addArrangedSubview(paymentMethodButton)
-                    }
-                    
-                case .goCardlessMandate:
-                    paymentMethodButton.setTitleColor(theme.colorTheme.text1, for: .normal)
-                    paymentMethodButton.tintColor = theme.colorTheme.tint1
-                    paymentMethodButton.layer.borderWidth = 1.0
-                    paymentMethodButton.layer.borderColor = theme.colorTheme.text1.cgColor
-//                    verticalStackView.addArrangedSubview(paymentMethodButton)
-                    
-                case .klarna:
-                    paymentMethodButton.backgroundColor = UIColor(red: 1, green: 0.702, blue: 0.78, alpha: 1)
-                    paymentMethodButton.setTitleColor(.black, for: .normal)
-                    paymentMethodButton.tintColor = .white
-                    paymentMethodButton.setImage(nil, for: .normal)
-                    paymentMethodButton.addTarget(self, action: #selector(klarnaButtonTapped), for: .touchUpInside)
-                    verticalStackView.addArrangedSubview(paymentMethodButton)
-                    
-                default:
-                    break
-                }
+        let checkoutViewModel: VaultCheckoutViewModelProtocol = DependencyContainer.resolve()
+        var availablePaymentMethods = checkoutViewModel.availablePaymentOptions.filter({ $0.type != .applePay })
+        
+        for (index, paymentMethod) in availablePaymentMethods.enumerated() {
+            switch paymentMethod.type {
+            case .klarna:
+                availablePaymentMethods[index].surCharge = "+£5.36"
```

Comment:
Hardcoded data? How do you plan to get this data from /configuration?

Reply:
Yeah, I've mentioned that on the description, I didn't have data back then. This has changed on the other branch that I was working with the local backend.